Fixes 7790: Add OpenCollective to list of allowed payment providers

CuriousLearner opened the pull request.

EnTeQuAk commented on src/olympia/constants/base.py:
```diff
@@ -222,7 +222,7 @@
     'donate.mozilla.org', 'micropayment.de', 'patreon.com', 'paypal.com',
-    'paypal.me',
+    'paypal.me', 'opencollective.com'
 )
```
We should start sorting this alphabetically, to be honest, otherwise, this looks good.

CuriousLearner commented:
@EnTeQuAk I've addressed your reviews & here is the screenshot of the changes on front-end:

<img width="721" alt="screen shot 2018-03-20 at 9 57 13 pm" src="https://user-images.githubusercontent.com/8039608/37668211-f4c29450-2c89-11e8-8142-b2643e1c538f.png">

EnTeQuAk commented:
Looks great, thanks a lot for the update!
